extensions/amp-script/0.1/amp-script.js

```diff
@@ -1,4 +1,4 @@
-import * as WorkerDOM from '@ampproject/worker-dom/dist/amp-production/main.mjs';
+import * as WorkerDOMImport from '@ampproject/worker-dom/dist/amp-production/main.mjs';
```

Comment:
Have you checked to see if this change, in particular assigning it to an object, might be causing the compiler to create it as an object and make all references into object accesses? I've seen that happen before with `import * as` syntax. The amp-script JS/MJS both went up by 70~80 bytes, while everything else fluctuated in normal range (<10 bytes). Given that this should in theory be a no-op change, since the new function should only ever be called by test code and thus DCE'd out of any binary, any bundle size increase is a red flag

Reply:
You're right, thats exactly what happened:
```
// Prev Definition:
function Eb(a,b,c){return c.hydrateFilter=Db...

// Prev callsite:
return Eb(b||this.element,l,d).

// WorkerDOM definition now:
var Eb={upgrade:function(a,b,c){return c.hydrateFilter=Db...

// Call site:
return Eb.upgrade(b||this.element,l,d)
```

That said, I don't know how to get around this?

Reply:
Stop trying to treat `import * as ns` as an object (a value assigned to variable, that can be mutated, can be introspected), and instead treat it as a namespace (an ambient lookup table). Every time 
you see `ns` in the code, it should exactly match `ns.PROP` get access.

Or, just get rid of the namespace entirely:

```js
import { upgrade } from "worker-dom";

let upgradeForTest = null;
function setUpgradeForTest(fn) {
  upgradeForTest = fn;
}

function layoutCallback() {
  …
  return (upgradeForTest || upgrade)( … );
}

Reply:
> Stop trying to treat import * as ns as an object

tbf, it was preexisting 😬 . point taken though!

Checking now to see if this pattern minifies properly.
If so, then this is viable!

Reply:
It was preexisting, but it only became a problem when you assigned it to a variable: `let upgrade = upgradeImported`. Any attempt to do that, whether in a function or not, is going to tell the compiler that `ns` needs to be an object/value that can be assigned, so it's going to make it into one.

Reply:
Doesn't happen with the `upgrade` case, only when the whole NS was assigned to a var